### PR TITLE
Stop silently failing on certain web socket connect errors

### PR DESCRIPTION
**Pull request by stkem**
\\+ removes forgiving client version extraction (now strict again)
@2is10 @jaredpetker 
This should probably only be merged once android changes are confirmed, so @jaredpetker you give the thumbs up when ready.

### CODE REVIEW

**Comment by jaredpetker** [comment]
I've made changes myself, of course any other clients out there may
suffer...should be fine though.

**Comment by stkem** [comment]
Cool. Given that it's relatively late in the day and this has the potential of causing a bunch of airbrakes I'll merge it in the morning.

File: kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala [hunk 1]
```diff
@@ -165,33 +165,34 @@ trait AuthenticatedWebSocketsController extends ElizaServiceController {
...
+              startMessages = startMessages :+ Json.arr("version", "new")
             }
+            onConnect(socketInfo)
+            (iteratee(streamSession, versionOpt, socketInfo, channel), Enumerator(startMessages: _*) >>> enumerator)
           }
-          val ipOpt: Option[String] = eipOpt.flatMap { eip =>
-            crypt.decrypt(ipkey, eip).toOption
+          iterateeAndEnumeratorFuture.onFailure {
+            case t: Throwable =>
+              airbrake.notify("Fatal error when establishing websocket connection", t)
```
Comment (2is10) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala:193]: I see. So Play closes the connection automatically on a failed `Future`, and this just ensures we report it. Feels a little odd that bad requests can trigger Airbrakes, but I guess only authenticated requests make it to this code path, so it’s better to assume we’ve made an error than to assume a user is playing with request parameters. Thanks!

**Comment by stkem** [comment]
Deploy in progress. Please let me know if you notice issues @jaredpetker

**Comment by stkem** [comment]
We are getting a bunch of errors fro iOS clients with a bad version. I'm rolling this back until we can confirm that these are being rejected legitimately (in an interview right now).

**Pull request merged**